frame/support/src/traits/voting.rs

```diff
@@ -20,81 +20,10 @@
 
 use crate::dispatch::{DispatchError, Parameter};
 use codec::{HasCompact, MaxEncodedLen};
-use sp_arithmetic::{
-	traits::{SaturatedConversion, UniqueSaturatedFrom, UniqueSaturatedInto},
-	Perbill,
-};
+use sp_arithmetic::Perbill;
 use sp_runtime::traits::Member;
 use sp_std::prelude::*;
 
-/// A trait similar to `Convert` to convert values from `B` an abstract balance type
-/// into u64 and back from u128. (This conversion is used in election and other places where complex
-/// calculation over balance type is needed)
-///
-/// Total issuance of the currency is passed in, but an implementation of this trait may or may not
-/// use it.
-///
-/// # WARNING
-///
-/// the total issuance being passed in implies that the implementation must be aware of the fact
-/// that its values can affect the outcome. This implies that if the vote value is dependent on the
-/// total issuance, it should never ber written to storage for later re-use.
-pub trait CurrencyToVote<B> {
```

Comment:
Wondering why this was moved to staking primitives? This is not only specific to staking right?

Reply:
We only use it in staking and elections pallet. But the main reason is that ina follow up PR I want to use this trait in StakingInterface, which is in primitives

Reply:
Can we move this to sp-core or some other better place? I need this for pallet-elections-phragmen but I am not using staking at all and shouldn't need to import sp-staking.